azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/AbstractWebAppMojo.java

```diff
@@ -416,6 +398,30 @@ protected WebAppConfig getWebAppConfig() throws AzureExecutionException {
         return parser.getWebAppConfig();
     }
 
+    protected AzureAppService getOrCreateLibraryClient() throws AzureExecutionException {
+        try {
+            final MavenAuthConfiguration mavenAuthConfiguration = auth == null ? new MavenAuthConfiguration() : auth;
+            mavenAuthConfiguration.setType(getAuthType());
+            final AzureCredentialWrapper azureCredentialWrapper = MavenAuthUtils.login(session, settingsDecrypter, mavenAuthConfiguration);
+            if (Objects.isNull(azureCredentialWrapper)) {
+                return null;
+            }
+            final AzureProfile profile = new AzureProfile(azureCredentialWrapper.getTenantId(), getSubscriptionId(),
+                    convertEnvironment(azureCredentialWrapper.getEnv()));
+            final AzureResourceManager azureResourceManager = AzureResourceManager.configure()
+                    .authenticate(azureCredentialWrapper.getTokenCredential(), profile).withSubscription(getSubscriptionId());
+            return AzureAppService.auth(azureResourceManager);
+        } catch (AzureLoginException | AzureExecutionException e) {
+            throw new AzureExecutionException(e.getMessage());
+        }
+    }
+
+    protected AzureEnvironment convertEnvironment(com.microsoft.azure.AzureEnvironment environment) {
```

Comment:
why this conversion?

Reply:
As AzureTokenWrapper still use track 1 environment